app/org/sagebionetworks/bridge/dynamodb/DynamoSchedulePlanDao.java

```diff
@@ -88,39 +95,100 @@ public SchedulePlan getSchedulePlan(StudyIdentifier studyIdentifier, String guid
         if (plans.isEmpty()) {
             throw new EntityNotFoundException(SchedulePlan.class);
         }
-        return plans.get(0);
+        
+        plan = plans.get(0);
+        forEachCriteria(plan, scheduleCriteria -> loadCriteria(scheduleCriteria));
+        return plan;
     }
 
     @Override
     public SchedulePlan createSchedulePlan(StudyIdentifier studyIdentifier, SchedulePlan plan) {
-        checkNotNull(studyIdentifier, "StudyIdentifier is null");
-        checkNotNull(plan, "Schedule plan is null");
+        checkNotNull(studyIdentifier);
+        checkNotNull(plan);
         
         plan.setStudyKey(studyIdentifier.getIdentifier());
         plan.setGuid(BridgeUtils.generateGuid());
         plan.setModifiedOn(DateUtils.getCurrentMillisFromEpoch());
+        
+        forEachCriteria(plan, scheduleCriteria -> persistCriteria(scheduleCriteria));
         mapper.save(plan);
         return plan;
     }
 
     @Override
     public SchedulePlan updateSchedulePlan(StudyIdentifier studyIdentifier, SchedulePlan plan) {
-        checkNotNull(studyIdentifier, "StudyIdentifier is null");
-        checkNotNull(plan, "Schedule plan is null");
+        checkNotNull(studyIdentifier);
+        checkNotNull(plan);
         
         plan.setStudyKey(studyIdentifier.getIdentifier());
         plan.setModifiedOn(DateUtils.getCurrentMillisFromEpoch());
+        
+        forEachCriteria(plan, scheduleCriteria -> persistCriteria(scheduleCriteria));
         mapper.save(plan);
         return plan;
     }
 
     @Override
     public void deleteSchedulePlan(StudyIdentifier studyIdentifier, String guid) {
-        checkNotNull(studyIdentifier, "Study identifier is null");
-        checkArgument(StringUtils.isNotBlank(guid), "Plan GUID is blank or null");
+        checkNotNull(studyIdentifier);
+        checkArgument(isNotBlank(guid));
         
         SchedulePlan plan = getSchedulePlan(studyIdentifier, guid);
+        
+        forEachCriteria(plan, scheduleCriteria -> deleteCriteria(scheduleCriteria));
         mapper.delete(plan);
     }
+    
+    private void forEachCriteria(SchedulePlan plan, Function<ScheduleCriteria,Criteria> consumer) {
+        if (plan.getStrategy() instanceof CriteriaScheduleStrategy) {
+            CriteriaScheduleStrategy strategy = (CriteriaScheduleStrategy)plan.getStrategy();
+            for (int i=0; i < strategy.getScheduleCriteria().size(); i++) {
+                ScheduleCriteria scheduleCriteria = strategy.getScheduleCriteria().get(i);
+                
+                // Add the key... builder makes this verbose
+                scheduleCriteria = new ScheduleCriteria.Builder()
+                        .withSchedule(scheduleCriteria.getSchedule())
+                        .withCriteria(scheduleCriteria.getCriteria())
+                        .withKey(getKey(plan, i)).build();
```

Comment:
What's the goal here? If the goal is to change or create the criteria key, the copyCriteria() method should do the trick. If the goal is to pass both the key and the criteria to the consumer, Java 8 allows lambdas with multiple args. (You might have to create a functional interface to represent this, but ultimately it'll be cleaner.)

If the goal is the preserve the "schedule" field, as noted in another comment, since CriteriaScheduleStrategy currently isn't in use, so we can refactor this to not include the "schedule" field.